pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,47 @@
+package ipsec
+
+import (
+	"strconv"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+type ipsecAPIMode struct {
+	state     string
+	legacyAPI string // It is a string since we need a value for uninitailized state ("")
+}
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode ipsecAPIMode // initial values are empty strings
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
+	registerMetrics.Do(func() {
+		ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name: "cluster_network_ipsec_state",
+			Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+		}, []string{"mode", "is_legacy_api"})
+		legacyregistry.MustRegister(ipsecStateGauge)
+	})
+}
+
+func UpdateIpsecTelemetry(state string, isLegacyAPI bool) {
+	klog.V(5).Infof("IPsec mode(prev): %s(%s), isLegacyAPI(prev): %v(%s)",
+		state, recordedMode.state, isLegacyAPI, recordedMode.legacyAPI)
+	isLegacyAPIStr := strconv.FormatBool(isLegacyAPI)
+	RegisterIpsecMetrics()
+	if state != recordedMode.state || isLegacyAPIStr != recordedMode.legacyAPI {
```

Comment:
It is not after upgrade - the API could be invoked from `ipsec: {}` to `ipsec.mode: "Full"`
from a functional perspective, it may not matter but I believe it is good for metrics purposes. Especially since there is no way back from the new API to the legacy one.